src/deploy.rs

```diff
@@ -77,4 +155,173 @@ impl Cmd {
         )?;
         Ok(())
     }
+
+    async fn run_against_rpc_server(&self, contract: Vec<u8>) -> Result<(), Error> {
+        let base_url = self.rpc_server_url.as_ref().unwrap().clone() + "/api/v1/jsonrpc";
+        // TODO: We should consider migrating the server subcommand to jsonspree
+        let client = HttpClientBuilder::default().build(base_url)?;
+        let key = parse_private_key(self.private_strkey.as_ref().unwrap())?;
+
+        // Get the account sequence number
+        let public_strkey =
+            stellar_strkey::StrkeyPublicKeyEd25519(key.public.to_bytes()).to_string();
+        let account_details: GetAccountResponse = client
+            .request("getAccount", rpc_params![public_strkey])
+            .await?;
+        // TODO: create a cmdline parameter for the fee isntead of simply using the minimum fee
+        let fee: u32 = 100;
+        let sequence = account_details.sequence.parse::<i64>()?;
+        let tx = build_create_contract_tx(
+            contract,
+            sequence,
+            fee,
+            self.network_passphrase.as_ref().unwrap(),
+            &key,
+        )?;
+        let response: SendTransactionResponse = client
+            .request("sendTransaction", rpc_params![tx.to_xdr_base64()?])
+            .await?;
+        println!("{}", response.status);
+        Ok(())
```

Comment:
Should probably poll `getTransactionStatus` to wait for the `"status": "success"`

Reply:
Ah, so `sendTransaction` isn't synchronous?

What's the meaning/point of the status field in the `sendTransaction` response then?

Reply:
@paulbellamy I implemented it, PTAL.